common/src/main/java/com/wynntils/features/inventory/BulkBuyFeature.java

```diff
@@ -83,26 +81,27 @@ public void onTooltipPre(ItemTooltipRenderEvent.Pre event) {
     private List<Component> replacePrices(List<Component> oldLore) {
         if (!KeyboardUtils.isShiftDown()) return oldLore;
 
-        CodedString priceLine = ComponentUtils.getCoded(oldLore.get(oldLore.size() - 1));
-        Matcher priceMatcher = priceLine.getMatcher(PRICE_PATTERN);
-        if (!priceMatcher.find()) {
-            WynntilsMod.warn("Could not find price for " + oldLore.get(0).getString() + " in " + priceLine);
-            return oldLore;
-        }
-        int newPrice = Integer.parseInt(priceMatcher.group(1)) * bulkBuyAmount.get();
-
-        CodedString newLine = CodedString.fromString(priceLine
-                .getInternalCodedStringRepresentation()
-                .replace(priceMatcher.group(1), BULK_BUY_ACTIVE_COLOR + Integer.toString(newPrice)));
-
-        if (newPrice > Models.Emerald.getAmountInInventory()) {
-            newLine = CodedString.fromString(newLine.getInternalCodedStringRepresentation()
-                    .replace("a✔", "c✖")); // Replace green checkmark with red x
+        // iterate through lore to find the price, then replace it with the bulk buy price
+        // there is no better way to do this since we cannot tell which line is the price (user may or may not have nbt
+        // lines enabled/disabled)
+        for (Component line : oldLore) {
+            StyledText oldLine = StyledText.fromComponent(line);
+            Matcher priceMatcher = oldLine.getMatcher(PRICE_PATTERN);
+            if (!priceMatcher.find()) continue;
+
+            int newPrice = Integer.parseInt(priceMatcher.group(1)) * bulkBuyAmount.get();
+            StyledText newLine = StyledText.fromString(oldLine.getInternalCodedStringRepresentation()
+                    .replace(priceMatcher.group(1), BULK_BUY_ACTIVE_COLOR + Integer.toString(newPrice)));
+            if (newPrice > Models.Emerald.getAmountInInventory()) {
+                newLine = StyledText.fromString(newLine.getInternalCodedStringRepresentation()
+                        .replace("a✔", "c✖")); // Replace green checkmark with red x
+            }
+            List<Component> newLore = new ArrayList<>(oldLore);
+            newLore.set(newLore.indexOf(line), newLine.getComponent());
+            return newLore;
```

Comment:
Can you refactor this so we only return at the end of the method (so we store the newLore list with a default value outside of the for loop)?